### PR TITLE
Implements #140: Add a native CSS framework

### LINKED ISSUE #140: CSS framework.
Currently we are using Bootstrap as the CSS framework but it is not being used widely across the application. also for proper function of bootstrap, we have to include the `jQuery` which is not quite necessary in the application. As angular in itself should be used for DOM manipulation not any external framework.

We can remove the framework and go on with either no framework approach or go with `angular/material2` though this project is in alpha now we can implement this later in 1-2 months.

* [ ] Remove Bootstrap
* [ ] Remove jQuery
* [ ] Fix the places where the UI breaks after applying the changes.

### PR DESCRIPTION
**Pull request by uttu357**
- Remove Bootstrap and JQuery
- Fix the places where the UI breaks

Closes #140

### CODE REVIEW

**Comment by uttu357** [comment]
@hemantjadon Please review the changes.. :)

File: src/styles/partials/_positioning.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
+h1, h2, h3 {
```
Comment (hemantjadon) [inline src/styles/partials/_positioning.scss]: Why these overrides?

File: src/styles/partials/_theme.scss [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+hr {
+	margin-top: 20px;
+	margin-bottom: 20px;
+	border: 0;
+	border-top: 1px solid $line-color;
+}
+
+
+
+/****HOME PAGE SEARCH BOX ELEMENTS****/
```
Comment (hemantjadon) [inline src/styles/partials/_theme.scss]: No styles  specific to home page should be in partials

File: src/styles/partials/_theme.scss [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+	border: 1px solid transparent;
+	display: inline-block;
+	padding: 6px 12px;
+	vertical-align: middle;
+	border-radius: 4px;
+	border-top-left-radius: 0px;
+	border-bottom-left-radius: 0px;
+}
+
+/****NAVBAR ELEMENTS****/
```
Comment (hemantjadon) [inline src/styles/partials/_theme.scss]: Similarly in navbar elements.

File: src/styles/partials/_theme.scss [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+				a {
+					padding-top: 15px;
+					padding-bottom: 15px;
+				}
+			}
+		}
+	}
+}
+
+/****ICONS****/
```
Comment (hemantjadon) [inline src/styles/partials/_theme.scss]: No need of using custom SVGs you can simply use Material icons which are already being included, in the project.

File: src/app/contact/contact.component.html [hunk 5]
```diff
@@ -1,28 +1,25 @@
...
-          <span class="sr-only">Toggle navigation</span>
-          <span class="icon-bar"></span>
-          <span class="icon-bar"></span>
-          <span class="icon-bar"></span>
-        </button>
+        <label for="navbar-toggle-cbox" type="button" class="navbar-toggle">
+          <div class="list-icon"></div>
+        </label>
         <a class="navbar-brand" routerLink="/">
 					<img src="../../assets/images/logo.png" alt="loklak social media search engine">
```
Comment (hemantjadon) [inline src/app/contact/contact.component.html]: Just noticed this no default `width` and `height` present please add this.

File: src/styles/main.scss [hunk 6]
```diff
@@ -0,0 +1,6 @@
+@import 'partials/reset';
```
Comment (hemantjadon) [inline src/styles/main.scss]: Shouldn't this be `import 'partials/_reset';` and also in all the lines.

**Comment by hemantjadon** [comment]
I think I am liking this idea of partials files. We can think of writing CSS related to specific to buttons and inputs there.

File: src/styles/main.scss (hunk 6 above)
Comment (uttu357) [inline src/styles/main.scss]: no they work that way http://thesassway.com/beginner/how-to-structure-a-sass-project

File: src/styles/partials/_theme.scss (hunk 3 above)
Comment (uttu357) [inline src/styles/partials/_theme.scss]: These are default styles for navbar which are same for multiple pages such as about, terms and contact and thus should be located in global styles?

File: src/styles/main.scss (hunk 6 above)
Comment (hemantjadon) [inline src/styles/main.scss]: Okay i don't know much if its working then its cool. 👍

File: src/styles/partials/_positioning.scss (hunk 1 above)
Comment (uttu357) [inline src/styles/partials/_positioning.scss]: These overrides are basic margins for every heading element and paragraph which were previously given by bootstrap to have a uniform styles over each page. They can be overridden in each component scss file if need be.

File: src/styles/partials/_theme.scss (hunk 2 above)
Comment (uttu357) [inline src/styles/partials/_theme.scss]: I forgot to remove HOME PAGE from comments. These styles correspond to the basic styling of both the search input at the home page and also on the header of search results page. Hence global. Only the basic styles regarding display and flow are here. The customisation according to the component are present in their component.scss file.

File: src/styles/partials/_positioning.scss (hunk 1 above)
Comment (hemantjadon) [inline src/styles/partials/_positioning.scss]: Why don't stick to what browser provides, i guess more often than not we are going to override them in each case

File: src/styles/partials/_theme.scss (hunk 3 above)
Comment (hemantjadon) [inline src/styles/partials/_theme.scss]: Well I guess then move navbar to its own Module. like Footer (`<app-footer>`). With its own component (`<app-navbar>`) which will contain its own styles.

File: src/styles/partials/_theme.scss (hunk 4 above)
Comment (uttu357) [inline src/styles/partials/_theme.scss]: Sure will do that.. :+1:

File: src/app/contact/contact.component.html (hunk 5 above)
Comment (uttu357) [inline src/app/contact/contact.component.html]: Sure.  :+1:

File: src/styles/partials/_positioning.scss (hunk 1 above)
Comment (uttu357) [inline src/styles/partials/_positioning.scss]: ohk ill remove them.. :+1:

File: src/styles/partials/_theme.scss (hunk 3 above)
Comment (uttu357) [inline src/styles/partials/_theme.scss]: Sure will do.. :+1:

**Comment by uttu357** [comment]
Yeah global styling can be used efficiently to impose a uniform theme for the website.

**Comment by uttu357** [comment]
@hemantjadon have made the changes and added the navbar module. Please review :)

File: src/app/navbar/navbar.component.html [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+        </a>
+    	</div>
+      <div class="navbar-collapse">
+        <ul>
+          <li><a routerLink="/">Home</a></li>
+          <li><a routerLink="/about">About</a></li>
+          <li><a routerLink="/contact">Contact</a></li>
+        </ul>
+      </div>
+    <!--/.nav-collapse -->
```
Comment (hemantjadon) [inline src/app/navbar/navbar.component.html]: Remove these comments, No need of them.

File: src/app/navbar/navbar.component.scss [hunk 8]
```diff
@@ -0,0 +1,89 @@
+@import '../../styles/partials/variables';
+
+.navbar {
+	background-color: $background-color-navbar;
```
Comment (hemantjadon) [inline src/app/navbar/navbar.component.scss]: I don't think that there should be variable for navbar, as it is going to be used only here not anyplace else, so I guess you can add only the `rgb()` values, as this is not going to be used at anyplace else.

**Comment by uttu357** [comment]
Done.. :)

**Comment by hemantjadon** [comment]
Seems good to me for the start. Though changes may be required in future.

**Pull request merged**